Content.Server/GameObjects/Components/Body/Surgery/BiologicalSurgeryDataComponent.cs

```diff
@@ -1,31 +1,125 @@
 
-        private readonly List<IMechanism> _disconnectedOrgans = new();
+        private readonly HashSet<IMechanism> _disconnectedOrgans = new();
 
-        private bool _skinOpened;
-        private bool _skinRetracted;
-        private bool _vesselsClamped;
+        private bool SkinOpened { get; set; }
 
-        protected override SurgeryAction? GetSurgeryStep(SurgeryType toolType)
+        private bool SkinRetracted { get; set; }
+
+        private bool VesselsClamped { get; set; }
+
+        public IBodyPart? Parent => Owner.GetComponentOrNull<IBodyPart>();
+
+        public BodyPartType? ParentType => Parent?.PartType;
+
+        private void AddDisconnectedOrgan(IMechanism mechanism)
+        {
+            if (_disconnectedOrgans.Add(mechanism))
+            {
+                Dirty();
+            }
+        }
+
+        private void RemoveDisconnectedOrgan(IMechanism mechanism)
+        {
+            if (_disconnectedOrgans.Remove(mechanism))
+            {
+                Dirty();
+            }
+        }
+
+        private async Task<bool> SurgeryDoAfter(IEntity performer)
+        {
+            if (!performer.HasComponent<DoAfterComponent>())
+            {
+                return true;
+            }
+
+            var doAfterSystem = EntitySystem.Get<DoAfterSystem>();
+            var target = Parent?.Body?.Owner ?? Owner;
+            var args = new DoAfterEventArgs(performer, 3, target: target)
+            {
+                BreakOnUserMove = true,
+                BreakOnTargetMove = true
+            };
+
+            return await doAfterSystem.DoAfter(args) == DoAfterStatus.Finished;
+        }
+
+        public string GetDescription()
+        {
+            if (Parent == null)
+            {
+                return string.Empty;
+            }
+
+            var toReturn = "";
+
+            if (SkinOpened && !VesselsClamped)
+            {
+                // Case: skin is opened, but not clamped.
+                toReturn += Loc.GetString("The skin on {0:their} {1} has an incision, but it is prone to bleeding.\n",
+                    Owner, Parent.Name);
+            }
+            else if (SkinOpened && VesselsClamped && !SkinRetracted)
+            {
+                // Case: skin is opened and clamped, but not retracted.
+                toReturn += Loc.GetString("The skin on {0:their} {1} has an incision, but it is not retracted.\n",
+                    Owner, Parent.Name);
+            }
+            else if (SkinOpened && VesselsClamped && SkinRetracted)
+            {
+                // Case: skin is fully open.
+                toReturn += Loc.GetString("There is an incision on {0:their} {1}.\n", Owner, Parent.Name);
+                foreach (var mechanism in _disconnectedOrgans)
+                {
+                    toReturn += Loc.GetString("{0:their} {1} is loose.\n", Owner, mechanism.Name);
+                }
+            }
+
+            return toReturn;
```

Comment:
Stringbuilder

loc at the end would be faster but this isn't called a lot so not a huge deal.

Reply:
Done.